### ISSUE TITLE
Add interactive features to request and select symbols

### ISSUE THREAD

**Issue by SteveALee** [mulberrysymbols/mulberry-symbols#31]
The original straight-street website which held the Mulberry symbols provided several interactive features for managing the symbols

* A process aimed mainly at Speech and Language Therapists to request and review new symbols.
* Users could select a subset symbols by browsing, tag or category and download as a zip
* A set of symbol translations and ability to select the users prefered language (names, not localised symbols)
* A RESTful API encouraging innovative apps that use the symbols online

These were all dropped on the copyright assignment and the related move to mulberry-symbols.org. The idea was to keep everything light weight until new requirments and or funding arose. Plus the codebase had  an enourmous amount of technical debt.

Let's discuss what similar features might be useful and how to add them....

**Comment by SteveALee** [comment]
@gavinhenderson @eadraffan FYI

**Comment by gavinhenderson** [comment]
I feel like it might be useful to follow a similar model to [FontAwesome ](https://github.com/FortAwesome/Font-Awesome) when it comes to making these symbols available to developers for their innovate apps. I can go into more detail about how I see this working for us if that helps?

**Comment by SteveALee** [comment]
Do tell! It's a mature package and is popular so they are probably doing somehting right.

Made me think a npm package is probably as useful as an REST API these days. Especially as buildtime resource acquisition often beats run time (eg static sites)

**Comment by eadraffan** [comment]
Thanks so much and I am copying in Russell who has been linking up all the symbol sets. I need to understand the concept behind it all!!

Best wishes
E.A.
Sent from my mobile phone

On 13 Feb 2019, at 07:42, Steve Lee <notifications@github.com<mailto:notifications@github.com>> wrote:


Do tell! It's a mature package and is popular so they are probably doing somehting right.

Made me think a npm package is probably as useful as an REST API these days. Especially as buildtime resource acquisition often beats run time (eg static sites)

**Comment by gavinhenderson** [comment]
There would be 1 centralized symbols repo and its only concern is to add new symbols. Every time this repo is altered it would trigger releases of all of the distributions. 

There would then be multiple distribution versions such as Vanilla JS, React, Vue, and many more. They would all come in their own git repo and be totally automated. You can then expand your distributions really easily and not have to worry about keeping the symbols up to date. 

Personally I would rather get the symbols in an NPM package rather than a REST API if I was developing a website, but I think that might be personal preference more than anything.

**Comment by SteveALee** [comment]
@gavinhenderson Do we actually need version for different frameworks? I'd like to avoid that if required as even if autmated it's complexity to manage. Would a plain Module of the right type (ES now?) not work almost everywhere Web front end now a days. We could perhaps provide sample wrappers. Needs more discussion.

**Comment by gavinhenderson** [comment]
Yeah that is a good point. I think having simple wrappers for different frameworks does add value as it removes all work for the end developer. I think making the library super easy to use can really increase usage of the package.

That being said I agree with you it does add unnecessary complexity to the automation so they can be ignored for now and examples of how to use it in different frameworks would suffice. Then maybe they can be reconsidered once at a later date.

**Comment by SteveALee** [comment]
I just realised that now we have a web page of all the symbols you can individually download any symbols you like

* Open https://mulberrysymbols.org/assets/categories/categories.html - it's big so a slow download
* Right click on a symbol and select "Save Image" or equivalent depending on your web browser
* Change the filename from index.svg to whatever you want

It's not ideal as the symbols name gets lost but acts as a temp workaround.

**Comment by SteveALee** [comment]
It looks like this could be done in GlobalSymbols - see #30
